packages/web3-providers-ws/test/integration/web_socket_provider_integration.test.ts

```diff
@@ -0,0 +1,338 @@
+
+describeIf(getSystemTestProvider().includes('ws'))(
+	'WebSocketProvider - implemented methods',
+	() => {
+		let clientWsUrl: string;
+		let accounts: string[];
+		let webSocketProvider: WebSocketProvider;
+		let jsonRpcPayload: Web3APIPayload<EthExecutionAPI, 'eth_getBalance'>;
+		// helper function
+		let currentAttempt = 0;
+		const waitForOpenConnection = async (provider: WebSocketProvider, status = 'connected') => {
+			return new Promise<void>((resolve, reject) => {
+				const maxNumberOfAttempts = 10;
+				const intervalTime = 5000; // ms
+
+				const interval = setInterval(() => {
+					if (currentAttempt > maxNumberOfAttempts - 1) {
+						clearInterval(interval);
+						reject(new Error('Maximum number of attempts exceeded'));
+					} else if (provider.getStatus() === status) {
+						clearInterval(interval);
+						resolve();
+					}
+					// eslint-disable-next-line no-plusplus
+					currentAttempt++;
+				}, intervalTime);
+			});
+		};
+		beforeAll(async () => {
+			clientWsUrl = getSystemTestProvider();
+			accounts = await getSystemTestAccounts();
+		});
+		describe('tests', () => {
+			beforeEach(() => {
+				jsonRpcPayload = {
+					jsonrpc: '2.0',
+					id: 42,
+					method: 'eth_getBalance',
+					params: [accounts[0], 'latest'],
+				} as Web3APIPayload<EthExecutionAPI, 'eth_getBalance'>;
+				webSocketProvider = new WebSocketProvider(
+					clientWsUrl,
+					{},
+					{ delay: 1, autoReconnect: false, maxAttempts: 1 },
+				);
+				currentAttempt = 0;
+			});
+			afterEach(async () => {
+				if (webSocketProvider.getStatus() !== 'disconnected') {
+					// make sure we try to close the connection after it is established
+					await waitForOpenConnection(webSocketProvider);
+					webSocketProvider.disconnect();
+				}
+			});
+
+			describe('websocker provider tests', () => {
+				it('should connect', async () => {
+					await waitForOpenConnection(webSocketProvider);
+					expect(webSocketProvider).toBeInstanceOf(WebSocketProvider);
+					expect(webSocketProvider.getStatus()).toBe('connected');
+				});
+			});
+
+			describe('subscribe event tests', () => {
+				it('should subscribe on message', done => {
+					webSocketProvider.on(
+						'message',
+						(
+							error: Error | null,
+							result?: JsonRpcSubscriptionResult | JsonRpcNotification<any>,
+						) => {
+							if (error) {
+								done.fail(error);
+							}
+							expect(result?.id).toBe(jsonRpcPayload.id);
+							done();
+						},
+					);
+					webSocketProvider.request(jsonRpcPayload).catch(err => {
+						done.fail(err);
+					});
+				});
+
+				it('should subscribe on error', done => {
+					const errorMsg = 'Custom WebSocket error occured';
+					webSocketProvider.on('error', (err: any) => {
+						expect(err?.message).toBe(errorMsg);
+						done();
+					});
+
+					webSocketProvider['_wsEventEmitter'].emit(
+						'error',
+						new Web3WSProviderError(errorMsg),
+					);
+				});
+
+				// eslint-disable-next-line jest/expect-expect
```

Comment:
That's a valid catch by elint. There should not be any test without any expect. Please add what your expectations to the tests and remove this comment.